src/chatd.cpp

```diff
@@ -4454,7 +4579,8 @@ Idx Chat::msgIncoming(bool isNew, Message* message, bool isLocal)
             if (mHasMoreHistoryInDb)
             { //we have db history that is not loaded, so we determine the index
               //by the db, and don't add the message to RAM
-                idx = mDbInterface->getOldestIdx()-1;
+                assert(mDbInterface->getOldestIdx() != CHATD_IDX_INVALID);
+                idx = mDbInterface->getOldestIdx() - 1;
```

Comment:
Actually, now that we keep the oldest index updated in a variable, I wonder if we could simply do a
```suggestion
                idx = mOldestIdxInDb - 1;
```